invenio_communities/models.py

```diff
@@ -288,31 +288,35 @@ def add_record(self, record):
         key = current_app.config['COMMUNITIES_RECORD_KEY']
         record.setdefault(key, [])
 
-        assert self.id not in record[key]
-        record[key].append(self.id)
-        record[key] = sorted(record[key])
-
+        if self.has_record(record):
+            current_app.logger.warning(
```

Comment:
I'd keep it as warning, although this fallback is "safe", it should still indicate that something elsewhere is modifying record outside the API, which is most likely a bug or misuse.